src/dotty/tools/dotc/ast/tpd.scala

```diff
@@ -372,9 +372,12 @@ object tpd extends Trees.Instance[Type] with TypedTreeInfo {
       newArr("Generic").appliedToTypeTrees(typeArg :: Nil)
     else if (elemClass.isPrimitiveValueClass)
       newArr(elemClass.name.toString)
-    else
-      newArr("Ref").appliedToTypeTrees(
-        TypeTree(defn.ArrayOf(elemType)).withPos(typeArg.pos) :: Nil)
+    else {
+      val typeApplied = newArr("Ref").appliedToTypeTrees(typeArg :: Nil)
+      val classOfArg =
+        ref(defn.Predef_classOf).appliedToTypeTrees(typeArg :: Nil)
```

Comment:
> Should I fix clsOf so that it is reliable along the entire pipeline?

I guess it would be the way to go. It should check if the type it is given derives from `scala.Array`, if it does, only don't yet erase its type before erasure.